plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1581,7 +1581,12 @@ private void finishOptimize(ConnectorSession session, IcebergTableExecuteHandle
         Snapshot snapshot = requireNonNull(icebergTable.snapshot(optimizeHandle.snapshotId().get()), "snapshot is null");
         rewriteFiles.validateFromSnapshot(snapshot.snapshotId());
         commit(rewriteFiles, session);
-        transaction.commitTransaction();
+        try {
+            transaction.commitTransaction();
+        }
+        catch (RuntimeException e) {
+            throw new TrinoException(ICEBERG_COMMIT_ERROR, "Failed to commit during optimize", e);
```

Comment:
Thus, users can understand the failed cause without looking into the stacktrace:
```suggestion
            throw new TrinoException(ICEBERG_COMMIT_ERROR, "Failed to commit during optimize: " + firstNonNull(e.getMessage(), e), e);
```